driver: usdhc: add option to disable 1.8v

lgl88911 opened the pull request:
When 1.8V is disabled, sdhc can only
communicate at low speed. But this can
save the external circuit for switching
between 3.3V and 1.8V, which is very
practical in costdown scenarios.

Signed-off-by: Frank Li <lgl88911@163.com>

lgl88911 committed:
driver: usdhc: add option to disable 1.8v

When 1.8V is disabled, sdhc can only
communicate at low speed. But this can
save the external circuit for switching
between 3.3V and 1.8V, which is very
practical in costdown scenarios.

Signed-off-by: Frank Li <lgl88911@163.com>

lgl88911 commented:
For some hardware design, NVCC_SD0 of rt1052 is directly connected to 3.3V, without using VSELECT control, so 1.8V support needs to be disabled through software to avoid compatibility problems on UHS card.

lgl88911 commented on drivers/disk/Kconfig.sdmmc:
```diff
@@ -41,6 +41,13 @@ config SDMMC_USDHC
 	  File system on a SDHC card accessed over
 	  NXP USDHC.
 
+config SDMMC_USDHC_DISABLE_V180
```
Although this is to solve the problem of hardware design. But I think this is close to the configuration of the software, because no matter how the external hardware is designed, the SDCard can only work under 3.3V through this configuration item.

jfischer-no replied:
I think something like this is better to maintain if it is configured via devicetree, I can also be wrong. @MaureenHelm , your opinion?

> Maybe default enable SDMMC_USDHC_DISABLE_V180 item through the devicetree? For example, add a bool attribute of vselect to nxp,imx-usdhc.yaml, Can you give some suggests? Thank you very much!

e.g. Linux Kernel binding for esdhc controller has voltage-ranges property, maybe that is the way to go,
https://www.kernel.org/doc/Documentation/devicetree/bindings/mmc/fsl-esdhc.txt

MaureenHelm replied:
Yes, I agree this should be configured via device tree.

lgl88911 replied:
@jfischer-no Thanks for your information, I found a more suitable node attribute
https://www.kernel.org/doc/Documentation/devicetree/bindings/mmc/mmc-controller.yaml
```
  no-1-8-v:
    $ref: /schemas/types.yaml#/definitions/flag
    description:
      When specified, denotes that 1.8V card voltage is not supported
      on this system, even if the controller claims it.
```

jfischer-no replied:
@lgl88911That looks good.

lgl88911 committed:
boards: mm_swiftio: fix sdcard power issue

In mm_swiftio hardware design, NVCC_SD0 of rt1052 is directly connected
to 3.3V, without using VSELECT control, so 1.8V support needs to be
disabled by no-1-8-v of devicetree.

Signed-off-by: Frank Li <lgl88911@163.com>

MaureenHelm commented on drivers/disk/usdhc.c:
```diff
@@ -2822,6 +2832,11 @@ static int disk_usdhc_init(const struct device *dev)
 		    (DISK_ACCESS_USDHC_INIT_NONE(n)))
 
+#define DISK_ACCESS_USDHC_INIT_NO_1_8_V(n)				\
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(n, no_1_8_v),			\
```
You don't need to check HAS_PROP for a boolean property. You can remove `DISK_ACCESS_USDHC_NO_1_8_V_PROPS` and `DISK_ACCESS_USDHC_INIT_NO_1_8_V`, and initialize the member unconditionally.

lgl88911 replied:
Thanks, Done

MaureenHelm commented on drivers/disk/usdhc.c:
```diff
@@ -2339,8 +2339,8 @@ static int usdhc_sd_init(struct usdhc_priv *priv)
-	* sdmmc will handle it automatically
-	*/
+	 * sdmmc will handle it automatically
+	 */
```
stray change

lgl88911 replied:
Thanks, Done

jfischer-no commented on dts/bindings/mmc/nxp,imx-usdhc.yaml:
```diff
@@ -27,5 +27,12 @@ properties:
+      type: boolean
+      required: false
+      description:
+        When specified, denotes that 1.8V card voltage is not supported
```
I think description should be phrased differently. This is taken from https://www.kernel.org/doc/Documentation/devicetree/bindings/mmc/mmc-controller.yaml (SPDX-License-Identifier: GPL-2.0)

lgl88911 replied:
Thanks for review, modify it.
